snakefmt/snakefmt.py

```diff
@@ -271,9 +271,11 @@ def main(
         LogConfig.switch()
         if diff or compact_diff:
             filename = "stdin" if path_is_stdin else str(path)
-            click.echo(f"{'=' * 5}> Diff for {filename} <{'=' * 5}\n")
-            difference = differ.compare(original_content, formatted_content)
-            click.echo(difference)
+            is_changed = differ.is_changed(original_content, formatted_content)
+            if is_changed:
+                difference = differ.compare(original_content, formatted_content)
+                click.echo(f"{'=' * 5}> Diff for {filename} <{'=' * 5}\n")
+                click.echo(difference)
```

Comment:
What's the expected output of `--diff` - the diff for changed files, yes, but should the user also be told the number of unchanged files, as for `--check`? If not, this is good. If yes, perhaps we `log.info` or `log.debug` the number of unchanged files?

Reply:
Well `--check` and `--diff` can be used together, so you can get the numbers by using both together and no numbers if you use `--diff` on it's own.